strings.go

```diff
@@ -0,0 +1,368 @@
+package main
+
+import (
+	"go/ast"
+	"go/token"
+	"log"
+	"strconv"
+	"strings"
+
+	"golang.org/x/tools/go/ast/astutil"
+)
+
+func obfuscateStrings(files []*ast.File) []*ast.File {
+
+	rmConst := func(cursor *astutil.Cursor) bool {
+		node := cursor.Node()
+
+		t, ok := node.(*ast.GenDecl)
+		if !ok {
+			return true
+		}
+
+		if t.Tok == token.CONST {
+			t.Tok = token.VAR
+		}
+
+		return true
+	}
+
+	var (
+		key        = genAesKey()
+		fset       = token.NewFileSet()
+		addedToPkg bool // we only want to inject the code and imports once
+	)
+
+	obfusStrings := func(cursor *astutil.Cursor) bool {
+		node := cursor.Node()
+
+		v, ok := node.(*ast.File)
+		if ok && !addedToPkg {
+			v.Decls = append(v.Decls, funcStmt())
+			v.Decls = append(v.Decls, keyStmt(key))
+
+			if v.Imports == nil {
+				var newDecls = []ast.Decl{
+					cryptoAesImportSpec,
+				}
+
+				for _, decl := range v.Decls {
+					newDecls = append(newDecls, decl)
+				}
+
+				v.Decls = newDecls
+			} else {
+				astutil.AddImport(fset, v, "crypto/aes")
+				astutil.AddImport(fset, v, "crypto/cipher")
+			}
+
+			addedToPkg = true
+
+			return true
+		}
+
+		if !(cursor.Name() == "Values" || cursor.Name() == "Rhs") {
+			return true // we don't want to obfuscate literals in Print Functions etc.
+		}
+
+		lit, ok := node.(*ast.BasicLit)
+		if !ok {
+			return true
+		}
+
+		if lit.Kind != token.STRING {
+			return true // we only want to obfuscate strings for now
+		}
+
+		value := litToString(lit.Value)
+
+		ciphertext, err := encAes([]byte(value), key)
+		if err != nil {
+			log.Println("Could not encrypt string:", err)
+			return true
+		}
+
+		cursor.Replace(ciphertextStmt(ciphertext))
+
+		return true
+	}
+
+	for _, file := range files {
+		file = astutil.Apply(file, rmConst, obfusStrings).(*ast.File)
+	}
+
+	return files
+}
+
+func litToString(lit string) string {
+	if strings.HasPrefix(lit, "\"") {
+		return strings.Trim(lit, "\"")
+	}
+
+	if strings.HasPrefix(lit, "`") {
+		return strings.Trim(lit, "`")
+	}
+
+	return lit
+}
+
+// ast definitions for injection
+var (
+	aesCipherStmt = &ast.AssignStmt{
+		Lhs: []ast.Expr{
+			&ast.Ident{
+				Name: "block",
+			},
+			&ast.Ident{
+				Name: "_",
+			},
```

Comment:
make all these `ast.Ident` literals a single line, to reduce verbosity

Reply:
Done